lib/docker.js

```diff
@@ -74,12 +87,14 @@ function promisedSpawn(args, options) {
     });
 
     if (options.useErrHandler || options.errMessage) {
-        promise = promise.catch(function(code, ret) {
+        promise = promise.catch(function(err) {
             if (options.errMessage) {
                 console.error('ERROR: ' + options.errMessage.split("\n").join("\nERROR: "));
             }
-            console.error('ERROR: ' + args.slice(0, 2).join(' ') + ' exited with code ' + code);
-            process.exit(code);
+            console.error('ERROR: ' + args.slice(0, 2).join(' ')
+                + ' exited with code ' + err.code
+                + ' and message ' + err.message);
```

Comment:
The message may be empty in some cases (user-induced errors), so this needs a guard.